packages/flutter/lib/src/painting/rrect_utils.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'basic_types.dart';
+import 'edge_insets.dart';
+
+/// Utility methods for manipulating [RRect] objects.
+abstract final class RRectUtils {
```

Comment:
I don't know that we need a new class and file for these methods. I was thinking more along the lines of a mixin. :) That could be defined in box_border.dart. Also, the docs should be clearer about the use case and how radius is incorporated into this. The docs should also link to the associated classed that are using it.

Reply:
I’ve moved the shared methods into borders.dart as a public RRectUtils class so both BoxBorder and TableBorder (in separate libraries) can use them without duplicating code.
Since they are in different libraries, a library-private helper wasn’t accessible from both.
I also added complete Dartdoc describing the use case, radius handling, and related classes, per the review feedback.

Reply:
I agree! Since EdgeInsets class already has similar functions like `inflateRect` and `deflateRect`, placing these methods within the same class would improve cohesion.
I'll update the PR to move the RRect utility methods to the EdgeInsets class, keeping them consistent with the existing naming pattern.